pyproject.toml

```diff
@@ -0,0 +1,80 @@
+[project]
+name = "neo"
+version = "0.12.0.dev0"
+
+authors = [{name = "Neo authors and contributors"}]
+
+description = "Neo is a package for representing electrophysiology data in Python, together with support for reading a wide range of neurophysiology file formats"
+
+readme = "README.rst"
+requires-python = ">=3.7"
+license = {text = "BSD 3-Clause License"}
+classifiers = [
+    "Programming Language :: Python :: 3",
+    "Operating System :: OS Independent",
+    "Intended Audience :: Science/Research",
+    "Natural Language :: English",
+    "Topic :: Scientific/Engineering",
+]
+
+dependencies = [
+    "packaging",
+    "numpy>=1.18.5",
+    "quantities>=0.12.1",
+]
+
+[project.urls]
+"Homepage" = "https://neuralensemble.org/neo"
+
+[build-system]
+requires = ["setuptools>=62.0"]
+build-backend = "setuptools.build_meta"
+
+[tool.setuptools]
+packages = ["neo"]
+package-dir = {"neo" = "neo"}
+
+[project.optional-dependencies]
+
+test = [
+    "pytest",
+    "pytest-cov",
+    
+    # datalad   # this dependency is covered by conda (environment_testing.yml)
+    "scipy>=1.0.0",
+    "pyedflib",
+    "h5py",
+    "igor",
+    "klusta",
+    "tqdm",
+    "nixio",
+    "matplotlib",
+    "ipython",
+    "coverage",
+    "coveralls",
+    "pillow",
+    "sonpy",
+    "pynwb",
+    "probeinterface",
+]
+
+docs = [
+    "docutils<0.18",
+]
+
+igorproio = ['igor']
+kwikio = ['klusta']
+neomatlabio = ['scipy>=1.0.0']
+nixio = ['nixio>=1.5.0']
+stimfitio = ['stfio']
+tiffio = ['pillow']
+edf = ['pyedflib']
+ced = ['sonpy']
+nwb = ['pynwb']
+maxwell = ['h5py']
+biocam = ['h5py']
+
+# TODO [all]
```

Comment:
@JuliaSprenger please note that I have not find a way to do the all the moment.

Reply:
I think for implementing the 'all' keyword in this framework we need to duplicate the requirements here. It's not nice, but acceptable I think.